packages/emotion-server/src/index.js

```diff
@@ -15,12 +15,10 @@ export function extractCritical(html) {
       ids[match[1]] = true
     }
   }
-
-  o.rules = sheet.sheet.slice().filter(x => {
-    RGX.lastIndex = 0
-    let match = RGX.exec(x)
-    const ret = match == null || ids[match[1]] || false
-    return ret
+  
+  o.rules = sheet.sheet.filter(x => {
+    const match = x.match(RGX)
```

Comment:
Are you sure we want to remove the `RGX.lastIndex = 0` line? This is a global regex and the index needs to reset after every run from my understanding.

Reply:
that's right for the way it was used before `RGX.exec(x)`, however it's not applied when using `String.prototype.match`

Reply:
👍 missed that part 😅